jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -86,47 +87,13 @@ static JarType determineJarType(Path jarPath) throws IOException {
    */
   static List<FileEntriesLayer> createExplodedModeLayersForStandardJar(
       Path jarPath, Path tempDirPath) throws IOException {
-    Path localExplodedJarRoot = tempDirPath;
-    ZipUtil.unzip(jarPath, localExplodedJarRoot);
     List<FileEntriesLayer> layers = new ArrayList<>();
 
-    // Get dependencies from Class-Path in the jar's manifest and add a layer each for non-snapshot
-    // and snapshot dependencies. If Class-Path is not present in the jar's manifest then skip
-    // adding the dependencies layers.
-    String classPath = null;
-    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
-      classPath = jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.CLASS_PATH);
-    }
-    if (classPath != null) {
-      Predicate<String> isSnapshot = name -> name.contains("SNAPSHOT");
-      List<String> allDependencies = Splitter.onPattern("\\s+").splitToList(classPath.trim());
-      List<Path> nonSnapshotDependencies =
-          allDependencies
-              .stream()
-              .filter(isSnapshot.negate())
-              .map(Paths::get)
-              .collect(Collectors.toList());
-      List<Path> snapshotDependencies =
-          allDependencies.stream().filter(isSnapshot).map(Paths::get).collect(Collectors.toList());
-      if (!nonSnapshotDependencies.isEmpty()) {
-        FileEntriesLayer.Builder nonSnapshotDependenciesLayerBuilder =
-            FileEntriesLayer.builder().setName(DEPENDENCIES);
-        nonSnapshotDependencies.forEach(
-            path ->
-                nonSnapshotDependenciesLayerBuilder.addEntry(
-                    path, APP_ROOT.resolve(RelativeUnixPath.get("dependencies")).resolve(path)));
-        layers.add(nonSnapshotDependenciesLayerBuilder.build());
-      }
-      if (!snapshotDependencies.isEmpty()) {
-        FileEntriesLayer.Builder snapshotDependenciesLayerBuilder =
-            FileEntriesLayer.builder().setName(SNAPSHOT_DEPENDENCIES);
-        snapshotDependencies.forEach(
-            path ->
-                snapshotDependenciesLayerBuilder.addEntry(
-                    path, APP_ROOT.resolve(RelativeUnixPath.get("dependencies")).resolve(path)));
-        layers.add(snapshotDependenciesLayerBuilder.build());
-      }
-    }
+    // Add dependencies layers.
+    layers.addAll(getDependenciesLayers(jarPath));
```

Comment:
In this case, I think it's fine to have
```suggestion
    List<FileEntriesLayer> layers = getDependenciesLayers(jarPath);
```